segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java

```diff
@@ -189,7 +191,7 @@ public void initChannel(SocketChannel ch) {
                      p.addLast(TLSHelper.TLS_HANDLER_NAME, sslHandler);
                  }
 
-                 ServerConnectionInboundHandler lsh = new ServerConnectionInboundHandler();
+                 ServerConnectionInboundHandler lsh = new ServerConnectionInboundHandler(connectionTracker);
```

Comment:
This could just as easily go into the append processor.

Reply:
This makes testing a lot simpler. Instead of adding complicated checks in `AppendProcessorTest` to figure out when to pause or resume a connection, it's much simpler to just verify that the `adjustOutstandingBytes` method is invoked on this mock with a predetermined value.

Then in `ServerConnectionInboundHandler`, it's much easier to reason about pausing or resuming connections when passing in specific values to that method.

Reply:
Ah. What if the ConnectionTracker method instead of returning a boolean took in a Connection that is paused or resumed?

Reply:
I assume you don't imply that ConnectionTracker has an internal collection of all registered connections. That would make cleaning up a very important task, as we could easily introduce leaks.

Do you mean that `ConnectionTracker.adjustOutstandingBytes` takes in a `ServerConnection` object that it itself will pause or resume?

If I understand your suggestion properly:
- We keep `ServerConnection` as it is (in master) with a `pauseConnection` and `resumeConnection`
- In `AppendProcessor`, which already has a reference to the `ServerConnection`, invokes `ConnectionTracker.pauseOrResumeConnection(int deltaBytes, int connectionOutstandingBytes, ServerConnection c)` ?
- Modify `ConnectionTracker` accordingly.

This would mean the following:
- Add yet another arg to the 2 AppendProcessor's constructors, and plumb it in all the tests that reference it directly.
- It may make `AppendProcessorTest` a bit tricky since I can't estimate the `connectionOutstandingBytes` properly due to the processing being async internally.


This could be done. Is your goal here to not have `ServerConnection` keep track of its outstanding bytes and leave that to `AppendProcessor`?

Reply:
yes